compiler/rustc_privacy/src/lib.rs

```diff
@@ -242,6 +242,42 @@ where
                 // This will also visit substs if necessary, so we don't need to recurse.
                 return self.visit_projection_ty(proj);
             }
+            ty::Alias(ty::Inherent, data) => {
+                if self.def_id_visitor.skip_assoc_tys() {
+                    // Visitors searching for minimal visibility/reachability want to
+                    // conservatively approximate associated types like `Type::Alias`
+                    // as visible/reachable even if `Type` is private.
+                    // Ideally, associated types should be substituted in the same way as
+                    // free type aliases, but this isn't done yet.
+                    return ControlFlow::Continue(());
+                }
+
+                self.def_id_visitor.visit_def_id(
+                    data.def_id,
+                    "associated type",
+                    &DefPath { def_id: data.def_id, tcx },
```

Comment:
can you use DelayDm here?

Reply:
`visit_def_id` takes a `&dyn Display` but `DelayDm` can only be turned into a `DiagnosticMessage` and it doesn't impl `Display`.